src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -51,6 +56,45 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
         
     }
 
+    @Override
+    public ChallengeDTO updateChallenge(Long challengeId, ChallengeDTO challengeDTO) throws NotFoundException {
+
+        Long userId;
+
+        if(challengeDTO.getChallengeCategory() != null) {
+
+            Optional<Long> challengeCategoryOptional = Optional.ofNullable(challengeDTO.getChallengeCategory().getId());
```

Comment:
Nu are sens sa folosesti Optional in cazul asta. Poti face direct verificare pe baza prezentei id-ului in dto. Dupa care verifici daca exista id-ul respectiv in baza de date.